di-ipv-cimit-stub/lambdas/put-contra-indicators/src/main/java/uk/gov/di/ipv/core/putcontraindicators/PutContraIndicatorsHandler.java

```diff
@@ -1,23 +1,60 @@
 package uk.gov.di.ipv.core.putcontraindicators;
 
 import com.amazonaws.services.lambda.runtime.Context;
-import com.amazonaws.services.lambda.runtime.RequestHandler;
-import com.google.gson.Gson;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.message.StringMapMessage;
 import uk.gov.di.ipv.core.putcontraindicators.domain.PutContraIndicatorsRequest;
 import uk.gov.di.ipv.core.putcontraindicators.domain.PutContraIndicatorsResponse;
+import uk.gov.di.ipv.core.putcontraindicators.exceptions.CiPutException;
+import uk.gov.di.ipv.core.putcontraindicators.service.ContraIndicatorsService;
 
-public class PutContraIndicatorsHandler
-        implements RequestHandler<PutContraIndicatorsRequest, String> {
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class PutContraIndicatorsHandler implements RequestStreamHandler {
 
     private static final Logger LOGGER = LogManager.getLogger();
-    private static final Gson gson = new Gson();
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final ContraIndicatorsService cimitService;
+
+    public PutContraIndicatorsHandler() {
+        this.cimitService = new ContraIndicatorsService();
+    }
+
+    public PutContraIndicatorsHandler(ContraIndicatorsService cimitService) {
+        this.cimitService = cimitService;
+    }
 
     @Override
-    public String handleRequest(PutContraIndicatorsRequest event, Context context) {
-        LOGGER.info(new StringMapMessage().with("EVENT TYPE:", event.getClass().toString()));
-        return gson.toJson(PutContraIndicatorsResponse.builder().result("success").build());
+    public void handleRequest(InputStream input, OutputStream output, Context context)
+            throws IOException {
+        LOGGER.info(new StringMapMessage().with("Function invoked:", "PutContraIndicators"));
+        PutContraIndicatorsRequest putContraIndicatorsRequest;
+        PutContraIndicatorsResponse response;
+
+        try {
+            putContraIndicatorsRequest = mapper.readValue(input, PutContraIndicatorsRequest.class);
+        } catch (Exception ex) {
```

Comment:
```suggestion
        } catch (IOException ex) {
```
Might be worth being more specific with this catch

Reply:
Done.